### PR TITLE
I try to fix Packet with 850 charset processing error

### PR DESCRIPTION
**Pull request by ruditimmermans**

### CODE REVIEW

File: pkg/charset/CharsetManager_test.go [hunk 1]
```diff
@@ -24,6 +24,24 @@ func TestDecodeFromCP866(t *testing.T) {
 
 }
 
+func TestEncodeToCP866(t *testing.T) {
```
Comment (vit1251) [inline pkg/charset/CharsetManager_test.go:27]: You mean CP850 here?

File: pkg/charset/CharsetManager_test.go [hunk 2]
```diff
@@ -24,6 +24,24 @@ func TestDecodeFromCP866(t *testing.T) {
...
+
+	charsetManager := NewCharsetManager(nil)
+
+	var msgBody string = "Привет, мир!"
```
Comment (vit1251) [inline pkg/charset/CharsetManager_test.go:31]: You also should add source message here on CP850 to validate response.

File: pkg/charset/CharsetManager_test.go [hunk 3]
```diff
@@ -24,11 +24,11 @@ func TestDecodeFromCP866(t *testing.T) {
 
 }
 
-func TestEncodeToCP866(t *testing.T) {
+func TestEncodeToCP850(t *testing.T) {
 
 	charsetManager := NewCharsetManager(nil)
 
-	var msgBody string = "Привет, мир!"
+	var msgBody string = "Hello World!"
```
Comment (vit1251) [inline pkg/charset/CharsetManager_test.go:31]: Actually I does not well known about using language in CP850
But unittest should use CP850 specific char to check runes like 'ü' or 'å'.
Test with Latin words is not well check charset translation.
Could you write some message on specific language to check that special char translation.
Result byte array should contain Unicode representation of you message on decoding,
and CP850 charset on encoding.

**Comment by ruditimmermans** [comment]
Vitold,

I found these i dont knoz if that would be help?

https://www.ascii-codes.com/cp850.html

**Comment by vit1251** [comment]
For example, you may use source string similar "Hello, world" on Deutsche at CP850 charset like "Schönen Tag" and check that rune "ö" still stay readable.

**Comment by ruditimmermans** [comment]
I have been look more on this but i can not found anywhere the raw bytes in Dutch we say "Hallo Wereld" but i dont know how to get the RAW bytes from this....

**Comment by vit1251** [comment]
> I have been look more on this but i can not found anywhere the raw bytes in Dutch we say "Hallo Wereld" but i dont know how to get the RAW bytes from this....

You may open simple text document write message and open with `WinHex for Windows` system or `xxd in Linux`.

**Comment by ruditimmermans** [comment]
Ok thanks it should be fixed now ;)

**Comment by ruditimmermans** [comment]
Good news after fixing this no lost echomails anymore the ones i lost before are now there so it's fixed ;)

**Comment by vit1251** [comment]
> Good news after fixing this no lost echomails anymore the ones i lost before are now there so it's fixed ;)

I see the problem in that PR with change default encoding for the product. I think that this parameter should be moved to the configuration and made customizable. Since now, after applying this patch, the Cyrillic encoding will stop working, which will create a problem with СP866 messages.

Adding such functionality has not yet been part of my plan, but I think that it needs to be added and implemented. I can't say yet when I will.

**Pull request closed without merging**